modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSingleColumnTableCrossApiTest.java

```diff
@@ -0,0 +1,269 @@
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import org.apache.ignite.lang.ErrorGroups.Sql;
+import org.apache.ignite.sql.SqlException;
+import org.apache.ignite.table.KeyValueView;
+import org.apache.ignite.table.RecordView;
+import org.apache.ignite.table.Table;
+import org.apache.ignite.table.Tuple;
+import org.apache.ignite.tx.IgniteTransactions;
+import org.apache.ignite.tx.Transaction;
+import org.apache.ignite.tx.TransactionOptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Checks basic operations on a single column table.
+ */
+public class ItSingleColumnTableCrossApiTest extends ClusterPerClassIntegrationTest {
+    private static final String TABLE_NAME = "test";
+
+    private static final String[] ENGINES = {"aipersist", "aimem", "rocksdb"};
+
+    @Override
+    protected int nodes() {
+        return 1;
+    }
+
+    @AfterEach
+    @Override
+    public void tearDown(TestInfo testInfo) throws Exception {
+        for (String engine : ENGINES) {
+            sql("delete from " + tableName(engine));
+        }
+
+        tearDownBase(testInfo);
+    }
+
+    @BeforeAll
+    public void beforeAll() {
+        for (String engine : ENGINES) {
+            String testZoneName = ("test_zone_" + engine).toUpperCase();
+
+            sql(String.format("create zone %s engine %s with partitions=1, replicas=3;", testZoneName, engine));
+            sql(String.format("create table %s (ID int primary key) with primary_zone='%s'", tableName(engine), testZoneName));
```

Comment:
We can have tables with the same name for different zones???? Something is wrong...